org.adempiere.base/src/org/compiere/util/Trx.java

```diff
@@ -131,7 +131,15 @@ public static String createTrxName (String prefix)
 			prefix = "Trx";
 		prefix += "_" + UUID.randomUUID(); //System.currentTimeMillis();
 		//create transaction entry
-		Trx.get(prefix, true);
+		Trx trx = Trx.get(prefix, true);
+		StackTraceElement[] st = new Throwable().fillInStackTrace().getStackTrace();
+		for (StackTraceElement ste : st) {
+			if (! Trx.class.getName().equals(ste.getClassName())) {
+				String displayName = ste.getClassName().concat("_").concat(ste.getMethodName());
+				trx.setDisplayName(displayName);
+				break;
+			}
+		}

```

Comment:
this could be expensive...

Reply:
Thanks @hengsin - added the SysConfig TRX_AUTOSET_DISPLAY_NAME (defaults to false).
So, it needs to be set to Y in order to assign the display name based on stack trace.

And also changed the scope, this is just done for transactions without a prefix set.